File: backend-server/application/src/main/java/com/apitable/workspace/entity/DatasheetRecordEntity.java


package com.apitable.workspace.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * Workbench - Datasheet Record Table.
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(keepGlobalPrefix = true, value = "datasheet_record")
public class DatasheetRecordEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * Primary key.
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * Operation ID.
     */
    private String recordId;

    /**
     * Datasheet ID(link#xxxx_datasheet#dst_id).
     */
    private String dstId;

    /**
     * Data recorded in one row (corresponding to each field).
     */
    private String data;

    /**
     * The historical version number sorted is the revision of the original operation, and the array subscript is the revision of the current record.
     */
    private String revisionHistory;

    /**
     * Version No.
     */
    private Long revision;

    /**
     * Field Update Information.
     */
    private String fieldUpdatedInfo;

    /**
     * Delete tag(0:No,1:Yes).
     */
    @TableLogic
    private Boolean isDeleted;

    /**
     * Create User.
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * Last Update User.
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * Create time.
     */
    private LocalDateTime createTime;

    /**
     * Update time.
     */
    private LocalDateTime updateTime;

    /**
     * Create Time.
     */
    private LocalDateTime createdAt;

    /**
     * Update Time.
     */
    private LocalDateTime updatedAt;

}

File: packages/room-server/src/database/datasheet/entities/datasheet.record.entity.ts
import { IRecordCellValue, IRecordMeta } from '@apitable/core';
import { BaseEntity } from 'shared/entities/base.entity';
import { Column, Entity } from 'typeorm';

/**
 * Workbench-Datasheet Record
 */
@Entity('datasheet_record')
export class DatasheetRecordEntity extends BaseEntity {
  @Column({
    name: 'record_id',
    nullable: true,
    comment: 'record ID',
    length: 50,
  })
  recordId!: string;

  @Column({
    name: 'dst_id',
    nullable: true,
    comment: 'datasheet ID(related#datasheet#dst_id)',
    length: 50,
  })
  dstId!: string;

  @Column('json', {
    name: 'data',
    nullable: true,
    comment: 'data recorded by a line (corresponding to each field)',
  })
  data?: IRecordCellValue;

  @Column({
    name: 'revision_history',
    nullable: true,
    comment: 'revisions of the original operations, sorted by revision, indices are revisions of the record',
    length: 5000,
    default: () => '0',
  })
  revisionHistory?: string;

  @Column({
    name: 'revision',
    nullable: true,
    comment: 'revision',
    unsigned: true,
    default: () => 0,
  })
  revision!: number;

  @Column('json', {
    name: 'field_updated_info',
    nullable: true,
    comment: 'field(cell) update information',
  })
  recordMeta?: IRecordMeta;
}
